examples/HikariExample/src/main/java/software/amazon/HikariFailoverExample.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon;
+
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+import software.amazon.jdbc.ds.AwsWrapperDataSource;
+
+/*
+An example to use the AWS Wrapper with Hikari with failover enabled.
+ */
+public class HikariFailoverExample {
+
+  private static final String USER = "username";
+  private static final String PASSWORD = "password";
+
+  public static void main(String[] args) throws SQLException {
+    HikariDataSource ds = new HikariDataSource();
+
+    // Configure the connection pool:
+    ds.setUsername(USER);
+    ds.setPassword(PASSWORD);
+
+    // Specify the underlying datasource for HikariCP:
+    ds.setDataSourceClassName(AwsWrapperDataSource.class.getName());
+
+    // Configure AwsWrapperDataSource:
+    ds.addDataSourceProperty("jdbcProtocol", "jdbc:postgresql:");
+    ds.addDataSourceProperty("databasePropertyName", "databaseName");
+    ds.addDataSourceProperty("portPropertyName", "portNumber");
+    ds.addDataSourceProperty("serverPropertyName", "serverName");
+
+    // The failover plugin throws failover-related exceptions that need to be handled explicitly by HikariCP,
+    // otherwise connections will be closed immediately after failover. Set `ExceptionOverrideClassName` to provide
+    // a custom exception class.
+    ds.setExceptionOverrideClassName("software.amazon.jdbc.util.HikariCPSQLException");
+
+    // Specify the driver-specific data source for AwsWrapperDataSource:
+    ds.addDataSourceProperty("targetDataSourceClassName", "org.postgresql.ds.PGSimpleDataSource");
+
+    // Configuring PGSimpleDataSource:
+    Properties targetDataSourceProps = new Properties();
+    targetDataSourceProps.setProperty("serverName", "db-identifier.cluster-XYZ.us-east-2.rds.amazonaws.com");
```

Comment:
In other examples these are grouped above where your USER and PASSWORD env vars are so that the user can see what they need to configure   
```
  // User configures connection properties here
public static final String POSTGRESQL_CONNECTION_STRING =
      "jdbc:aws-wrapper:postgresql://test-db.cluster-XYZ.us-east-2.rds.amazonaws.com:5432";
```
could also set property like: getDataSourceProps.setProperty("databaseName", DATABASE_NAME")  with DATABASE_NAME set above on line 33 as well

Reply:
we could, postgres is a default database that gets created when users create their server so the code would still run if they don't customize it.